### PR TITLE
Refactor "app/notification_email.go" to use structured logging #12726

### PR DESCRIPTION
**Pull request by arjitc**
#### Summary

Refactor "app/notification_email.go" to use structured logging #12726

#### Ticket Link

https://github.com/mattermost/mattermost-server/issues/12726

### CODE REVIEW

**Comment by arjitc** [comment]
@levb it seems to be failing at the "Check i18n" stage, I tried to re-run the check in jenkins but it errors with,

```
+ cd /go/src/github.com/mattermost/mattermost-server

/home/ubuntu/jenkins/workspace/attermost-server-pr-new_PR-12731@tmp/durable-cba8dbdf/script.sh: 2: cd: can't cd to /go/src/github.com/mattermost/mattermost-server

script returned exit code 2
```

**Comment by arjitc** [comment]
Made a few changes and it errors with this now,

```
+ cd /go/src/github.com/mattermost/mattermost-server

+ cp i18n/en.json /tmp/en.json

+ make i18n-extract

env GO111MODULE=off go get -u github.com/mattermost/mattermost-utilities/mmgotool

/go/bin/mmgotool i18n extract

panic: 322:110: missing ',' in argument list (and 9 more errors)
```

**Comment by cpanato** [comment]
@arjitc can you please merge latest master in this pr? thanks!

**Comment by arjitc** [comment]
@cpanato -- I'm sorry not too sure how to do that :/ Can you please help me with that?

**Comment by cpanato** [comment]
```bash
git fetch upstream --prune
git merge upstream/master
git push origin patch-4
```

**Comment by arjitc** [comment]
@cpanato done :+1:

File: app/notification_email.go [hunk 1]
```diff
@@ -320,7 +320,7 @@ func (a *App) GetMessageForNotification(post *model.Post, translateFunc i18n.Tra
 	// extract the filenames from their paths and determine what type of files are attached
 	infos, err := a.Srv.Store.FileInfo().GetForPost(post.Id, true, false, true)
 	if err != nil {
-		mlog.Warn(fmt.Sprintf("Encountered error when getting files for notification message, post_id=%v, err=%v", post.Id, err), mlog.String("post_id", post.Id))
+		mlog.Warn("Encountered error when getting files for notification message", mlog.String("post_id", post.Id) mlog.Err(err))
```
Comment (levb) [inline app/notification_email.go]: @arjitc you are still missing a comma here?

Comment (levb) [inline app/notification_email.go]:
```
# github.com/mattermost/mattermost-server/app
app/notification_email.go:323:110: syntax error: unexpected mlog, expecting comma or )
# github.com/mattermost/mattermost-server/app [github.com/mattermost/mattermost-server/app.test]
app/notification_email.go:323:110: syntax error: unexpected mlog, expecting comma or )
make: *** [Makefile:155: govet] Error 1
```

Comment (arjitc) [reply app/notification_email.go]: Oops! Sorry about that, all fixed now @levb  :+1:

**Comment by cpanato** [comment]
@arjitc please update your PR and resolve the conflits

**Comment by arjitc** [comment]
@cpanato done :+1: :)

**Comment by cpanato** [comment]
@arjitc need another PR update

**Comment by arjitc** [comment]
I think this PR can be marked as a duplicate/closed as it seems like someone else has pushed the same fixes since into master :)

1. https://github.com/mattermost/mattermost-server/blob/master/app/notification_email.go#L103
2. https://github.com/mattermost/mattermost-server/blob/master/app/notification_email.go#L288

**Comment by levb** [comment]
@arjitc I think your PR has the more conventional field names, cc @grundleborg just in case to confirm. However, you still need to sync&merge master into your branch, and then pass the tests before it's merged.

**Comment by grundleborg** [comment]
Yeah I agree with Lev.

**Comment by arjitc** [comment]
Done now, should be in-sync -- @levb @grundleborg :+1:

**Pull request merged**